### ISSUE TITLE
running the tests

### ISSUE THREAD

**Issue by psteinb** [bethgelab/foolbox#296]
I am having a hard time running the tests on a CentOS7 box with `pip3 install --user` and friends. I am constantly running into problems with numpy when trying to run `test_attacks_boundary.py`. I am running into [this](https://github.com/numpy/numpy/issues/11788) or [this](https://github.com/scikit-image/scikit-image/issues/3655).

I am wondering if anyone has a Dockerfile oder Singularity file available that would host the development environment for foolbox needed. I'd like to make at least one contribution for tracking the number of iterations in black-box attacks.

**Comment by psteinb** [comment]
Alright, I was able to run the tests inside a container. If people are interested, the `Dockerfile` is [here](https://github.com/psteinb/foolbox/tree/docker_container/docs/docker). The Dockerfile is based on the travis recipe used in this repo. 

It turns out that inside the container, only 1 test fails. Apart from that, all runs well. I then checked versions and made my local baremetal installation with pip3 compliant to what the container exposed. Calling `pytest` inside the repo again yielded error messages. I then did `pip3 install -e .` and ran the tests again ... all passed. So I am  guessing that it's `pip3 install -e .` That does the trick.

One thing I noticed, the test suite is quite monolithic in a sense that `conftest.py` imports many classes which have dependencies (e.g. `from foolbox.models import CaffeModel`). This imposes that in order to run ALL tests (as they all import `conftest.py`) require CaffeModel to be installed. I believe this is counter productive in attracting more contributors. I would suggest, to refactor that part out so that maybe `conftest.py` relies e.g. on keras+tensorflow mostly (or whatever your choose to be a sensible default).
